File: 80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0, j = 0, k = 0;
        while (i < nums.size()) {
            nums[k++] = nums[i];
            j = i + 1;
            while (j < nums.size() && nums[i] == nums[j])
                j++;
            if (j - i >= 2)
                nums[k++] = nums[i];
            i = j;
        }
        return k;
    }
};

File: go/src/leetcode/80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	last := 0
	count := 1
	n := len(nums)
	for i := 1; i < n; i++ {
		if nums[i] == nums[last] {
			if count >= 2 {
				continue
			} else {
				count++
				last++
				nums[last] = nums[i]
			}
		} else {
			last++
			nums[last] = nums[i]
			count = 1
		}
	}
	fmt.Println(nums[:last+1])
	return last
}

func main() {
	removeDuplicates([]int{1,1,1,2,2,3})
	removeDuplicates([]int{0,0,1,1,1,1,2,3,3})
}